functions/src/business-logics/comment-logics.ts

```diff
@@ -1,7 +1,111 @@
 // TODO write onCommentCreateLogic
-//  increment post commentsCount
-//  create a notification for post's author
+import {LogicConfig, LogicResultDoc } from "emberflow/lib/types";
+import { Entity } from "../db-structure";
+import { firestore } from "firebase-admin";
+import DocumentData = firestore.DocumentData;
+import {Notification, Post, UserView} from "../types";
+import { admin, db } from "emberflow/lib";
+// eslint-disable-next-line import/namespace
+import {createUserView} from "./utils";
 
+export const onCommentCreateLogic: LogicConfig = {
+  //  increment post commentsCount
+  //  create a notification for post's author
+  name: "onCommentCreateLogic",
+  actionTypes: ["create"],
+  modifiedFields: "all",
+  entities: [Entity.Comment],
+  logicFn: async (action) => {
+    const {
+      eventContext: {docId, docPath},
+      user,
+      modifiedFields,
+    } = action;
+
+    // increment post commentsCount
+    const commentRef = db.doc(docPath);
+    const postDocRef = commentRef.parent.parent;
+
+    if (!postDocRef) {
+      return {
+        name: "onCommentCreateLogic",
+        status: "error",
+        documents: [],
+        message: `Invalid docPath at ${docPath}`,
+      };
+    }
+
+    const postRef = db.doc(postDocRef.path);
+    const postDocSnapShot = await postRef.get();
+    const postDoc = postDocSnapShot.data();
+    if (postDoc === undefined) {
+      return {
+        name: "onCommentCreateLogic",
+        status: "error",
+        documents: [],
+        message: `post ${postDocRef.id} does not exist`,
+      };
+    }
+
+    const {
+      "@id": postId,
+      createdBy: {
+        "@id": postOwnerId,
+      },
+    } = postDoc;
+
+    const postLogicResultDoc: LogicResultDoc = {
+      action: "merge",
+      dstPath: `posts/${postId}`,
+      instructions: {
+        commentsCount: "++",
+      },
+    };
+
+    const userView: UserView = createUserView(user);
+    const now: firestore.Timestamp = admin.firestore.Timestamp.now();
+
+    const notificationDoc: Notification = {
+      "@id": docId,
+      createdBy: userView,
+      createdAt: now.toDate(),
+      type: "comment",
+      read: false,
+      post: postDoc as Post,
+    };
+
+    const notificationLogicResultDoc: LogicResultDoc = {
+      action: "create",
+      dstPath: `users/${postOwnerId}/notifications/${docId}`,
+      doc: notificationDoc,
+    };
+
+    const commentDoc: DocumentData = {
+      ...modifiedFields,
+      "@id": docId,
+      "createdBy": userView,
+      "createdAt": now,
+      "repliesCount": 0,
```

Comment:
HI Jeremy, I’d like you to add a new field called post to the document. This will be an object that serves as the view of the post, with a type of PostView. You can add this to types.ts.

Here's the interface for PostView:

export interface PostView extends BaseEntity {
    createdAt: Date;
    createdBy: UserView;
}


The goal is to avoid fetching the post document when deleting a comment. All necessary information will now be accessible within the actual document. You can refer to Noel's onCommentDeleteLogic for context

Reply:
done adding the post field in the comment interface o7